dbt/adapters/firebolt/impl.py

```diff
@@ -28,19 +28,32 @@ class FireboltIndexConfig(dbtClassMixin):
     def render_name(self, relation):
         """
         Name an index according to the following format, joined by `_`:
-        index type, relation name, key/join column, timestamp (unix & UTC)
+        index type, relation name, key/join columns, timestamp (unix & UTC)
         example index name: join_my_model_customer_id_1633504263.
         """
         now_unix = time.mktime(datetime.utcnow().timetuple())
-        spine_col = '_'.join(self.key_column if self.key_column else self.join_column)
+        # If column_names is a list with > 1 members, do a string join,
+        #  otherwise do not. We were getting index names like
+        # join__idx__emf_customers__f_i_r_s_t___n_a_m_e__165093112.
+        if self.key_column:
```

Comment:
I think we can simplify this part
Doing join on a list with a single element would result in that element
```
column_names = self.key_column or self.join_column
spine_col = '_'.join(column_names) if isinstance(column_names, list) else column_names
```

Reply:
Also, why are these fields called `*_column`, when they might contain a list?

Reply:
Yeah, that's a _lot_ cleaner.

I didn't name those variables, I inherited them, but it's a good point. I've updated the names.